pkg/controller/pxc/full_crash_recovery.go

```diff
@@ -0,0 +1,153 @@
+package pxc
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	v1 "github.com/percona/percona-xtradb-cluster-operator/pkg/apis/pxc/v1"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+var (
+	ErrNotAllPXCPodsRunning = errors.New("not all pxc pods are running")
+	logLinesRequired        = int64(8)
+)
+
+const logPrefix = `#####################################################LAST_LINE`
+
+func (r *ReconcilePerconaXtraDBCluster) recoverFullClusterCrashIfNeeded(cr *v1.PerconaXtraDBCluster) error {
+	if cr.Spec.PXC.Size <= 0 {
+		return nil
+	}
+
+	err := r.checkIfPodsRunning(cr)
+	if err != nil {
+		if err == ErrNotAllPXCPodsRunning {
+			return nil
+		}
+		return err
+	}
+
+	logOpts := &corev1.PodLogOptions{
+		Container: "pxc",
+		TailLines: &logLinesRequired,
+	}
+	logLines, err := r.clientcmd.PodLogs(cr.Namespace, cr.Name+"-pxc-0", logOpts)
+	if err != nil {
+		return errors.Wrap(err, "get logs from pxc 0 pod")
+	}
+
+	for _, log := range logLines {
+		if strings.HasPrefix(log, logPrefix) {
+			return r.doFullCrashRecovery(cr.Name, cr.Namespace, int(cr.Spec.PXC.Size))
+		}
+	}
+
+	return nil
+}
+
+func (r *ReconcilePerconaXtraDBCluster) isPodWaitingForRecovery(namespace, podName string) (bool, int64, error) {
+	logOpts := &corev1.PodLogOptions{
+		Container: "pxc",
+		TailLines: &logLinesRequired,
+	}
+	logLines, err := r.clientcmd.PodLogs(namespace, podName, logOpts)
+	if err != nil {
+		return false, -1, errors.Wrapf(err, "get logs from %s pod", podName)
+	}
+
+	for _, log := range logLines {
+		if strings.HasPrefix(log, logPrefix) {
+			return parseSequence(log)
+		}
+	}
```

Comment:
Should we start checking from the last line, not the first? Is it possible to have a node log with the old and new crash? I mean is it possible to have:
```
#####################################################FULL_PXC_CLUSTER_CRASH:#####################################################
You have the situation of a full PXC cluster crash...
#####################################################LAST_LINE:nodename:13:#####################################################
#####################################################FULL_PXC_CLUSTER_CRASH:#####################################################
You have the situation of a full PXC cluster crash...
#####################################################LAST_LINE:nodename:17:#####################################################
```

Reply:
It should not, since we are checking such logs on all nodes.
But this logs will be removed after restart(after previous crash recovery)
But yea, better to make this check

Reply:
fixed there